### PR TITLE
sync: support cancel_rx in ttrpc context

### PR DESCRIPTION
**Pull request by yuqitao**
Fixes: https://github.com/containerd/ttrpc-rust/issues/180

### CODE REVIEW

File: Cargo.toml
Comment (yuqitao) [reply Cargo.toml:23]: With crossbeam channel, we can use `select!` in ttrpc method handler.
like: 
```
select! {
       recv(ticker) -> _ => println!("elapsed: {:?}", start.elapsed()),
       recv(timeout) -> _ => break,
}
```

Comment (jsturtevant) [reply Cargo.toml:23]: interesting!  seems like a possibly good improvement thought I would tend towards bring in crossbeam as dep in separate PR across all the areas where   std::sync::mpsc::Receiver/Sender is used to keep it similiar throughout the codebase.  I will defer to the maintainers though.

File: src/sync/server.rs
Comment (yuqitao) [reply src/sync/server.rs]: done

**Comment by yuqitao** [comment]
@lifupan @jsturtevant PTAL.

**Review by lifupan — APPROVED** [review]

**Comment by yuqitao** [comment]
@Tim-Zhang The CI seems to have some wired error, could you help solving it please?

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks @yuqitao

**Pull request merged**
